### ISSUE TITLE
Some old assets have no version and therefore cannot be cloned

### ISSUE THREAD

**Issue by tinok** [kobotoolbox/kpi#3020]
Reproduction steps: 
1. Subscribe to someone else's collection 
2. open collection
3. for any question or block in the collection, click clone button
4. enter new name for the question

Expected behavior: 
Question (or block) is cloned into My Library (no parent collection). User is shown `#/library/asset/[assetid]`

Actual: 
I receive a 500 error ("could not create project")

**Comment by jnm** [comment]
This doesn't seem to happen on newer collections, but I can reproduce it on the IRC stuff from 2016, e.g. https://kf.beta.kbtdev.org/#/library/asset/a6fFed7bKYsFZ9nAAKRjc7

**Comment by jnm** [comment]
Old-school blocks have no version? https://sentry.kbtdev.org/kobo/kpi-backend-dev/issues/306442/?query=is%3Aunresolved
```
AttributeError: 'NoneType' object has no attribute 'name'
  File "django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "rest_framework/viewsets.py", line 114, in view
    return self.dispatch(request, *args, **kwargs)
  File "rest_framework/views.py", line 505, in dispatch
    response = self.handle_exception(exc)
  File "rest_framework/views.py", line 465, in handle_exception
    self.raise_uncaught_exception(exc)
  File "rest_framework/views.py", line 476, in raise_uncaught_exception
    raise exc
  File "rest_framework/views.py", line 502, in dispatch
    response = handler(request, *args, **kwargs)
  File "kpi/views/v2/asset.py", line 272, in create
    serializer = self._get_clone_serializer()
  File "kpi/views/v2/asset.py", line 706, in _get_clone_serializer
    cloned_data = self._prepare_cloned_data(original_asset, source_version, partial_update)
  File "kpi/views/v2/asset.py", line 727, in _prepare_cloned_data
    cloned_data = original_asset.to_clone_dict(version=source_version)
  File "kpi/models/asset.py", line 1012, in to_clone_dict
    'name': version.name,
```

**Comment by jnm** [comment]
```
>>> working = Asset.objects.get(uid='arm66Tmd679PNkTiA59P3q')
>>> failing = Asset.objects.get(uid='aikzyDGuXasMFKwRc2Nz6D')
>>> working.name
'GPS test'
>>> failing.name
'Camp Management'
>>> working.asset_versions.count()
2
>>> failing.asset_versions.count()
0
```

Also wondered if the problem could be that the assets were created via bulk import, but I imported `collection_small.xlsx` from #2784 and couldn't reproduce the problem with the assets created from that.

**Comment by jnm** [comment]
Yes, this applies only to old library items, with the exception of a handful of `survey`s:
```
>>> oops_no_ver = Asset.objects.filter(asset_versions=None)
>>> non_col_no_ver = oops_no_ver.exclude(asset_type='collection')
>>> non_col_no_ver.order_by().values('asset_type').annotate(Count('pk'))
<QuerySet [{'asset_type': 'question', 'pk__count': 22612}, {'asset_type': 'block', 'pk__count': 530}, {'asset_type': 'survey', 'pk__count': 6}]>
>>> non_col_no_ver.exclude(asset_type='survey').aggregate(Max('date_modified'))
{'date_modified__max': datetime.datetime(2016, 10, 17, 19, 49, 55, 800921, tzinfo=<UTC>)}
```
See internal discussion [here](https://chat.kobotoolbox.org/#narrow/stream/4-KoBo-Dev/topic/500.20error.20when.20cloning.20old.20library.20items.20.233020/near/17993) for the two users who own these 6 `survey`s

**Comment by jnm** [comment]
Closed by #3027
